src/libs/actions/Report.js

```diff
@@ -379,11 +379,26 @@ function setLocalLastRead(reportID, sequenceNumber) {
 
     // Update the report optimistically
     Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
-        unreadActionCount: 0,
+        unreadActionCount: Math.max(reportMaxSequenceNumbers[reportID] - sequenceNumber, 0),
         lastVisitedTimestamp: Date.now(),
     });
 }
 
+/**
+ * Set the New marker for a given report to the oldest unread message
+ *
+ * @param {Number} reportID
+ */
+function setNewMarkerInitialPosition(reportID) {
+    const lastReadSequenceNumber = allReports[reportID].unreadActionCount === 0
+        ? 0
```

Comment:
> why not just always use the lastReadsequenceNumber for the new marker position

Because, 3 seconds after the report opens (or any time a message comes in), that gets set to the very last message. At that point we do NOT want to move the marker.
Which is why we are setting `newMarkerSequenceNumber` separately, once, when the report is created

Reply:
>Because, 3 seconds after the report opens (or any time a message comes in), that gets set to the very last message. At that point we do NOT want to move the marker.

And so, we need this new value because otherwise it would be difficult to tell whether the previous "last read" should persist or not when it changes for some unknown reason?

Reply:
Function gone! 
We moved the math back to the View. Let me know if you still have concerns about it, because, tbh, I'm not sure I'm following what you are trying to accomplish here, but:

> But presumably when we have an unreadActionCount > 0 "position" should always be the "last read"?
Only when setting the position for the first time (when the report becomes visible). Afterwards, "last read" will get updated, but the marker position should not change (unless we manually change it).